src/state/user.ts

```diff
@@ -28,10 +28,19 @@ export enum Deadlines {
   Custom = "CUSTOM",
 }
 
+export type ConfirmModalType = {
+  open: boolean
+  options?: {
+    modalTitle?: string
+    modalText?: string
+    onOK?: () => void
```

Comment:
Are we doing anything to prevent this state from being stale? It seems like it's possible for Flow X to accidentally call onOK that was enqueued by Flow Y, which would be problematic. 

I think these should either be namespaced, or somehow cleared between flows (although multiple flows may exist on one page, which is another problem).

I think what you will end up with is some version of a Modal Controller with an interface similar to https://www.npmjs.com/package/react-modal-controller